tests/Integration/Behaviour/Features/Context/CarrierFeatureContext.php

```diff
@@ -274,23 +275,24 @@ public function getCarrierWithName($carrierName)
     /**
      * Be careful: this method REPLACES shipping fees for carrier
      *
-     * @Given /^carrier "(.+)" applies shipping fees of (\d+\.\d+) in zone "(.+)" for quantities between (\d+) and (\d+)$/
+     * @Given /^carrier "(.+)" applies shipping fees of (\d+\.\d+) in zone "(.+)" for (weight|price) between (\d+) and (\d+)$/
      */
-    public function setCarrierFees($carrierName, $shippingPrice, $zoneName, $fromQuantity, $toQuantity)
+    public function setCarrierFees($carrierName, $shippingPrice, $zoneName, $rangeType, $from, $to)
     {
         $this->checkCarrierWithNameExists($carrierName);
         $this->checkZoneWithNameExists($zoneName);
         if (empty($this->carriers[$carrierName]->getZone((int) $this->zones[$zoneName]->id))) {
             $this->carriers[$carrierName]->addZone((int) $this->zones[$zoneName]->id);
         }
-        $rangeId = RangePrice::rangeExist($this->carriers[$carrierName]->id, $fromQuantity, $toQuantity);
+        $rangeClass = $rangeType=='weight' ? RangeWeight::class : RangePrice::class;
+        $rangeId = $rangeClass::rangeExist($this->carriers[$carrierName]->id, $from, $to);
```

Comment:
$rangeClass::rangeExist() return the result of a COUNT() not an id

Reply:
indeed, my bad
thank you for the report ;)

Reply:
We should add phpdoc on this because there are many mistakes with this